File: src/main/java/com/jss/osiris/modules/reporting/service/UserReportingService.java
package com.jss.osiris.modules.reporting.service;

import java.util.List;

import com.jss.osiris.modules.profile.model.Employee;
import com.jss.osiris.modules.reporting.model.UserReporting;

public interface UserReportingService {
    public List<UserReporting> getUserReportings(Employee employee);

    public UserReporting getUserReporting(Integer id);

    public UserReporting addOrUpdateUserReporting(UserReporting userReporting);

    public void copyUserReportingToUser(UserReporting userReporting, Employee employee);
}

File: client/src/app/modules/reporting/services/user.reporting.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Employee } from '../../profile/model/Employee';
import { UserReporting } from '../../reporting/model/UserReporting';

@Injectable({
  providedIn: 'root'
})
export class UserReportingService extends AppRestService<UserReporting>{

  constructor(http: HttpClient) {
    super(http, "reporting");
  }

  getUserReportings(employee: Employee) {
    return this.getList(new HttpParams().set("employeeId", employee.id), "user-reportings");
  }

  getUserReporting(id: number) {
    return this.getById("user-reporting", id);
  }

  addOrUpdateUserReporting(userReporting: UserReporting) {
    return this.addOrUpdate(new HttpParams(), "user-reporting", userReporting, "Enregistré", "Erreur lors de l'enregistrement");
  }

  copyUserReportingToUser(userReporting: UserReporting, employee: Employee) {
    return this.get(new HttpParams().set("userReportingId", userReporting.id!).set("employeeId", employee.id), "user-reporting/copy", "Le rapport a été copié chez " + (employee.firstname + " " + employee.lastname), "Erreur lors de la copie");
  }

  deleteUserReporting(userReporting: UserReporting) {
    return this.get(new HttpParams().set("userReportingId", userReporting.id!), "user-reporting/delete");
  }

}
